pkg/mmesh/modelmesh_events.go

```diff
@@ -202,6 +202,10 @@ func (mes *ModelMeshEventStream) refreshWatches(nw *namespaceWatch, namespace, s
 					if hashIdx > ownerIdx && key[hashIdx] == '-' {
 						// Infer predictor/vmodel and source ids from concrete model id by removing hash suffix
 						sourceId, predictorName := key[ownerIdx:hashIdx], key[:ownerIdx-2]
+						i := strings.LastIndex(namespace, "_")
+						if i > 0 {
+							namespace = namespace[i+1:]
+						}
```

Comment:
It went into a loop of adding sourceId + "_" without this.

Reply:
Are you sure that's still the case? There was an oversight in the first version of my PR that caused this, but it was fixed before we merged it. Could you try taking out these changes to verify it's still a problem?
If so then there's another bug somewhere we should fix instead of making these changes.